packages/plugin-assets/src/upload.js

```diff
@@ -0,0 +1,253 @@
+const ora = require('ora');
+const inquirer = require('inquirer');
+const { TwilioServerlessApiClient } = require('@twilio-labs/serverless-api');
+const { TwilioCliError } = require('@twilio/cli-core').services.error;
+const {
+  getEnvironment,
+} = require('@twilio-labs/serverless-api/dist/api/environments');
+const {
+  createAssetResource,
+  listAssetResources,
+  createAssetVersion,
+} = require('@twilio-labs/serverless-api/dist/api/assets');
+const {
+  triggerBuild,
+  activateBuild,
+  waitForSuccessfulBuild,
+  getBuild,
+} = require('@twilio-labs/serverless-api/dist/api/builds');
+
+const { resolve, basename } = require('path');
+const { readFile } = require('fs').promises;
+const EventEmitter = require('events');
+
+const {
+  listFunctionResources,
+} = require('@twilio-labs/serverless-api/dist/api/functions');
+
+const spinner = ora();
+
+const upload = async ({
```

Comment:
Can you break the different steps in this function into individual functions? Just makes reading this and debugging it easier.